processing/src/main/java/org/apache/druid/query/context/ResponseContext.java

```diff
@@ -44,9 +47,47 @@
 public abstract class ResponseContext
 {
   /**
-   * Keys associated with objects in the context.
+   * The base interface of a response context key.
+   * Should be implemented by every context key.
    */
-  public enum Key
+  public interface BaseKey
+  {
+    String getName();
+    /**
+     * Merge function associated with a key: Object (Object oldValue, Object newValue)
+     */
+    BiFunction<Object, Object, Object> getMergeFunction();
+  }
+
+  /**
+   * Keys associated with objects in the context. The enum is extension-friendly.
```

Comment:
I think it doesn't make a lot of sense to say that "The enum is extension-friendly." Enum is not extension-friendly, but the key system (based from `BaseKey`) is. So I would just remove this sentence.

Reply:
Removed